lib/api/documentExtractor.js

```diff
@@ -139,9 +168,18 @@ const extractors = ([
         for (let it = 0; it < documents.length; it++) {
           const document = documents[it];
 
-          request.input.body.documents.push({
-            _id: document._id,
-            body: document._source });
+          if (request.input.action === 'mUpsert') {
+            request.input.body.documents.push({
+              _id: document._id,
+              changes: document._source.changes,
```

Comment:
Same comment here you should only have a `_source` property containing the changes